crates/bevy_asset/src/server/mod.rs

```diff
@@ -494,16 +494,14 @@ impl AssetServer {
     ///
     /// `input_handle` must only be [`Some`] if `should_load` was true when retrieving `input_handle`. This is an optimization to
     /// avoid looking up `should_load` twice, but it means you _must_ be sure a load is necessary when calling this function with [`Some`].
-    async fn load_internal<'a>(
+    async fn load_internal_owned(
         &self,
         mut input_handle: Option<UntypedHandle>,
-        path: AssetPath<'a>,
+        path: AssetPath<'static>,
```

Comment:
This change doesn't accomplish anything. Lifetime generics are erased, not monomorphized.

Reply:
I agree that it _shouldn't_ matter, but it provably reduces the final binary size by 118KiB, more than what I could reasonably chalk up to compiler noise. If you look at [the commit](https://github.com/bevyengine/bevy/pull/14934/commits/b5249249e1597c5d196d6cfffb23aa6f9c4fb32f) in isolation you can see in the description the `cargo bloat` analysis I did before and after just that one change.

I still don't have a thorough explanation as to _why_ this reduces binary size so drastically, since the `'a` lifetime parameter is purely for the borrow checker. My only guess is that because this is an `async fn`, the state machine generation must be dependent on whether this value can be assumed `'static` or not.

Reply:
Note that the commit didn't just change a lifetime but also moved a function call outside the function, to another async function, which I think is what actually causes the reduction in size (though I'm not sure **why**, maybe a different allocation pattern for the Future locals?).

After that change the `'static` lifetime is necessary due to some functions being called requiring it, but it shouldn't impact code generation though, since that that stage the compiler already erased all informations about specific lifetimes.

Reply:
Yes that is also a valid point, since it shouldn't be possible for the lifetime parameter to contribute to code bloat, it's reasonable to question whether moving that one function call (which also shouldn't contribute to code bloat) is also part of the explanation. I do think this has something to do with how the `async` side of the function works, since I haven't seen any non-async functions with only lifetime parameters contribute to code bloat.